test/transitive_dependency/CMakeLists.txt

```diff
@@ -22,16 +22,23 @@ CPMAddPackage(
   OPTIONS "CXXOPTS_BUILD_EXAMPLES Off" "CXXOPTS_BUILD_TESTS Off"
 )
 
+option(BUILD_SHARED_LIBS "Create shared libraries" YES)
+if(BUILD_SHARED_LIBS)
+  # Set default visibility to hidden for all targets
+  set(CMAKE_CXX_VISIBILITY_PRESET hidden)
+  set(CMAKE_VISIBILITY_INLINES_HIDDEN YES)
+endif()
+
```

Comment:
Wouldn't it make sense to define these options explicitly from outside vs adding an option?

Reply:
Yes, but then on GitHub CI both variants, static and shared must be tested.

Reply:
You can set these unconditionally, CMake will just ignore the properties when they do not apply.